src/Microsoft.Data/DataFrame.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// A DataFrame 
+    /// </summary>
+    public partial class DataFrame
+    {
+        internal DataFrameTable _table;
+        public DataFrame()
+        {
+            _table = new DataFrameTable();
+        }
+
+        public long RowCount => _table.RowCount;
+        public int ColumnCount => _table.ColumnCount;
+        public IList<string> Columns()
+        {
+            var ret = new List<string>();
+            for (int ii = 0; ii < ColumnCount; ii++)
+            {
+                ret.Add(_table.Column(ii).Name);
+            }
+            return ret;
+        }
+        public BaseDataFrameColumn Column(int index) => _table.Column(index);
+        public void InsertColumn(int columnIndex, BaseDataFrameColumn column) => _table.InsertColumn(columnIndex, column);
+        public void SetColumn(int columnIndex, BaseDataFrameColumn column) => _table.SetColumn(columnIndex, column);
+        public void RemoveColumn(int columnIndex) => _table.RemoveColumn(columnIndex);
+        public void RemoveColumn(string columnName) => _table.RemoveColumn(columnName);
+        
+        public object this[long rowIndex, int columnIndex]
```

Comment:
Use expression bodied form for oneliners
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties#expression-body-definitions
eg
``c#
        public object this[long rowIndex, int columnIndex]
        {
            get => _table.Column(columnIndex)[rowIndex];
            set =>_table.Column(columnIndex)[rowIndex] = value;
        }
```
easier to read/shorter